### PR TITLE
[#3007] Add common commands into the synopsis

### LINKED ISSUE #3007: Put a summary of the common basic commands at the top of the inline help
This proposal for the inline CLI documentation proposal is inspired by [chrisnevers's remark on
discuss](https://discuss.ocaml.org/t/mkocaml-release-project-generator/4949/3?u=shonfeder):

> I just want to know the bare minimum commands to build/run/publish my project.

I think this is how a lot of users feel about their build tools, and in the
spirit of well designed, deep API's, this is all most users should need to
know.

Currently, the output of `dune`/`dune --help` is its manual page: the
first substantial piece of information are two paragraphs about the design
philosophy and lineage of `dune`, which is followed by an alphabetical listing
of the commands. This is great for a manual page, but not helpful for quick
guidance on general usage.

I propose foregrounding the few commands that all users need to know at the top
of the inline help. 

These are the three options I've thought of so far:

1. Follow `cargo`'s example and provide a list the commands with a synopsis of what they do. E.g.,
    ```
    Common dune commands (run *dune CMD --help* for more on the CMD):
      build    Build the given targets, or all installable targets if none are given.
      test     Run all tests in the current project
      exec     Build and execute a command or executable component in an environment as if it were installed
      init     Initialize a new project or component
    ```
    And the full list of commands could be given further down, as it is currently.
2. Demonstrate common usage by example at the top, E.g.,
    ```sh
    USAGE
      # Initialize a new project named my_project with executable and library components
      dune init proj my_project && cd my_project 
      dune build            # Build all targets
      dune test             # Run all tests
      dune exec my_project  # Run the executable named myproject
      dune install          # Install packages defined in the project
      dune clean            # Clean up all build artifacts
    ```
3. Follow `pacman`'s example, and list the synopsis of the most used commands. E.g.,
    ```
    Common operations:
      dune build [OPTION]... [TARGET]...
      dune runtest [OPTION]... [DIR]...
      dune exec [OPTION]... PROG [ARGS]...
      dune install [OPTION]... [PACKAGE]...
      dune clean [OPTION]...
      dune init [OPTION]... INIT_KIND NAME [PATH]
    ```

(1) has the benefit of providing a succinct summary of the commands. (2) Is more
verbose, and less general, but has the benefit of providing a mini tutorial on
a basic workflow. (3) Splits the differences, and could be augmented with some
commentary. 

If it is agreed that this is a good change to make, here are some things to do
moving forward:

- Consider other sources of inspiration or suggestions on how to approach this.
- Determine which commands are the most frequently used and/or most important
  for beginners to use.
- Pick a scheme.
- Make needed alterations to the cmdliner configuration

### PR DESCRIPTION
**Pull request by shonfeder**
This is a very minimal approach to resolve #3007. It relies on the
command names to be self-documenting, supported by the fact that more
extensive information is available just below.

It aims to be consistent with the purposes of the SYNPOSIS section
according to MAN(1) while stile providing useful hints to newer users.

2020-02-22-205544_1443x1090_scrot

### CODE REVIEW

**Comment by shonfeder** [comment]
Any other commands we should include?

**Comment by rgrinberg** [comment]
I don't think `dune clean` should be included. If you run this command regularly, you're doing something wrong. `dune install` is fairly useful on the other hand.

**Comment by diml** [comment]
This list seems good to me!

Just one comment: should we list `test` or `runtest`? From previous discussions, it seemed to me that we added `test` as an alias for `runtest` to make things easier for newcomers but without the intention to replace `runtest`.

I feel like to keep things consistent we should either:
- list `runtest` instead of `test` here
- list `test` and generally talk about `dune test` rather than `dune runtest` and mark `runtest` as a (possibly deprecated) alias for `test`

WDYT?

File: bin/main.ml [hunk 1]
```diff
@@ -195,6 +195,24 @@ let all =
   ; Caching.command
   ]
 
+let common_commands_synopsis =
+  (* Short reminders for the most used and useful commands *)
+  let commands =
+    [ "build [--wait]"
```
Comment (diml) [inline bin/main.ml]: Should be `--watch` (and bellow as well)

Comment (shonfeder) [reply bin/main.ml]: Doh! Thank you for catching that.

This confirms my thought that I should figure how how to make this testable...

**Comment by shonfeder** [comment]
Good point re: `test`/`runtest`. I'm in favor of using `runtest` here. I'm indifferent to which we settle on in the future, but happy to make the change throughout the codebase if we decide to standardize on `test`.

**Comment by shonfeder** [comment]
I fixed the three mentioned points.

Do either have you know off hand how I might put these commands under test? Is there an existing mechanism to share some strings between the blackbox tests and the CLI definition?

**Comment by diml** [comment]
To be fair, in this case testing seems more trouble than it is worth. Relying on code review seems enough to me in this case.

**Comment by shonfeder** [comment]
I agree testing isn't required here, but it would be clever if we could figure out how to it! I'm happy to leave that as a low priority improvement that will never be important enough to get addressed ;)

**Pull request merged**